packages/velo-external-db-commons/lib/data_commons.js

```diff
@@ -77,6 +79,20 @@ const extractProjectionFunctionsObjects = (projection) => projection.filter(f =>
 
 const isNull = (value) => (value === null || value === undefined)
 
+const specArrayToRegex = (spec) => {
+    if (!Array.isArray(spec)) {
+        throw new InvalidQuery('$matches must have array - spec property')
+    }
+    return spec.map(spec => {
```

Comment:
maybe extart the function in the map operator? 
I'm afraid that there are too many things happening the `specArrayToRegex` function